pkg/bootstrapper/bootstrapper.go

```diff
@@ -830,3 +828,58 @@ func (cni *cniOptions) configure(kubeletCmd *string) error {
 
 	return nil
 }
+
+// convertKubeletArgsToMap converts the list of kubelet arguements into a map. This is required
+// to reconstruct kubelet command when updating the kubelet service
+func convertKubeletArgsToMap(kubeletArgs []string) (map[string]string, error) {
+	kubeletKeyValueArgs := make(map[string]string)
+
+	for _, option := range kubeletArgs {
+		// Args like --config=c:\\k\\kubelet.conf will be split on "=" and stored as key value pairs of the map.
+		//Stand alone args like --windows-service will be stored as a string against a special key
+		if strings.Contains(option, "=") {
+			kv := strings.Split(option, "=")
+			kubeletKeyValueArgs[kv[0]] = kv[1]
+			// This is to account for args like --register-with-taints=os=Windows:NoSchedule
+			if len(kv) > 2 {
+				for _, val := range kv[2:] {
+					kubeletKeyValueArgs[kv[0]] += "=" + val
+				}
+			}
+		} else {
+			kubeletKeyValueArgs[kubeletStandAloneArgsKey] += option + " "
+		}
+	}
+
+	// Remove the trailing space
+	if standaloneArgs, found := kubeletKeyValueArgs[kubeletStandAloneArgsKey]; found {
+		kubeletKeyValueArgs[kubeletStandAloneArgsKey] = strings.TrimSpace(standaloneArgs)
+	}
+	return kubeletKeyValueArgs, nil
+
+}
+
+// configureKubelet updates the given kubelet command with the required args.
+func configureKubelet(kubeletCmd *string, kubeletArgs []string) error {
+	if kubeletCmd == nil {
+		return fmt.Errorf("nil kubelet cmd passed")
+	}
+	// Get kubelet args in key value format
+	kubeletKeyValueArgs, err := convertKubeletArgsToMap(kubeletArgs)
+	if err != nil {
+		return fmt.Errorf("unable to deconstruct kubelet command %s: %v", *kubeletCmd, err)
+	}
+
+	// Index 0 of kubeletArgs will hold the kubelet.exe.
+	kubeletPath := strings.Split(*kubeletCmd, " ")[0]
+	if !strings.Contains(kubeletPath, "kubelet.exe") {
+		return fmt.Errorf("kubelet command does not start with kubelet.exe")
+	}
+	kubeletKeyValueArgs[kubeletExeKey] = kubeletPath
+
+	if *kubeletCmd, err = reconstructKubeletCmd(kubeletKeyValueArgs); err != nil {
+		return fmt.Errorf("unable to reconstruct kubelet command %v: %v", kubeletKeyValueArgs, err)
+	}
```

Comment:
The benefit of converting the args into a map is that if we want to change specific arguments that is made easier. In this case you are not doing that, and it seems like you are creating a new function to convert a []string into a map so that the `reconstructKubeletCmd` can be reused. While its good to reuse functions, this is adding unnecessary logic and obfuscating a simple operation. 

Why was the below solution we talked about unviable?
```	
        s := syscall.EscapeArg(kubeletpath)
	for _, v := range args {
		s += " " + syscall.EscapeArg(v)
	}
```
This code was borrowed from the [svc mgr library](https://github.com/golang/sys/blob/master/windows/svc/mgr/mgr.go#L126-L129)

Reply:
I agree that we can get away without converting the []string to map, and can update that.

However I am not sure why the above code snippet is required when we can pass `&c.BinaryPath` where c is the `inital config`

Reply:
Its important to escape the arguments in case we are sourcing any problematic strings from the ignition file

Reply:
Discussed this, neither of the input is needed, we could generate the string and update the c.BinaryPath